src/envoy/utils/mixer_control.cc

```diff
@@ -100,5 +102,137 @@ Grpc::AsyncClientFactoryPtr GrpcClientFactoryForCluster(
   return std::make_unique<EnvoyGrpcAsyncClientFactory>(cm, service);
 }
 
+// create Local attributes object and return a pointer to it.
+// Should be freed by the caller.
+const LocalAttributes *createLocalAttributes(const localAttributesArgs &local) {
+  ::istio::mixer::v1::Attributes ib;
+  auto &inbound = (*ib.mutable_attributes());
+  inbound[AttributeName::kDestinationUID].set_string_value(local.uid);
+  inbound[AttributeName::kContextReporterUID].set_string_value(local.uid);
+  inbound[AttributeName::kDestinationNamespace].set_string_value(local.ns);
+  if (!local.ip.empty()) {
+    // TODO: mjog check if destination.ip should be setup for inbound.
+  }
+
+  ::istio::mixer::v1::Attributes ob;
+  auto &outbound = (*ob.mutable_attributes());
+  outbound[AttributeName::kSourceUID].set_string_value(local.uid);
+  outbound[AttributeName::kContextReporterUID].set_string_value(local.uid);
+  outbound[AttributeName::kSourceNamespace].set_string_value(local.ns);
+
+  ::istio::mixer::v1::Attributes fwd;
+  auto &forward = (*fwd.mutable_attributes());
+  forward[AttributeName::kSourceUID].set_string_value(local.uid);
+  return new LocalAttributes(ib, ob, fwd);
+}
+
+//
+// "sidecar~10.36.0.15~fortioclient-84469dc8d7-jbbxt.service-graph~service-graph.svc.cluster.local"
+//  --> {proxy_type}~{ip}~{node_name}.{node_ns}~{node_domain}
+bool extractInfo(localAttributesArgs *args, std::string nodeid) {
+  auto parts = StringUtil::splitToken(nodeid, "~");
```

Comment:
this makes me very sad...

Reply:
That is there only for compatibility with existing node names.

The other function uses node metadata.